### PR TITLE
Adding the ability to rename sites

### PR DESCRIPTION
**Pull request by cody-constine-ttd**
Before Rename:
image
After Rename:
image

### CODE REVIEW

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java [hunk 1]
```diff
@@ -388,6 +389,9 @@ private void handleSiteUpdate(RoutingContext rc) {
                     ResponseUtil.error(rc, 400, "Invalid parameter for visible: " + visibleParam);
                 }
             }
+            if(name != null) {
```
Comment (cYKatherine) [inline src/main/java/com/uid2/admin/vertx/service/SiteService.java:392]: Do we check if the new name is duplicated to the existing sites?

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java [hunk 2]
```diff
@@ -876,4 +876,25 @@ void addSiteWithDuplicateAppNames(Vertx vertx, VertxTestContext testContext) {
             testContext.completeNow();
         });
     }
+
+    @Test
```
Comment (cYKatherine) [inline src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:880]: Maybe add a test case when the newName is duplicated to the existing site names

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java (hunk 1 above)
Comment (cYKatherine) [reply src/main/java/com/uid2/admin/vertx/service/SiteService.java:392]: Also maybe a sanity check for empty string `name != ''`, we had a bug before where name was set to empty string and caused some hassle debugging :(

Comment (thomasm-ttd) [inline src/main/java/com/uid2/admin/vertx/service/SiteService.java:392]: Spacing around the if

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java [hunk 3]
```diff
@@ -876,4 +876,25 @@ void addSiteWithDuplicateAppNames(Vertx vertx, VertxTestContext testContext) {
...
+        setSites(sites);
+
+        post(vertx, testContext, "api/site/update?id=11&name=NewName", null, response -> {
+            assertEquals(200, response.statusCode());
+            assertEquals("NewName", response.bodyAsJsonObject().getString("name"));
+            assertEquals("NewName", sites[0].getName());
+            testContext.completeNow();
+        });
+    }
 }
```
Comment (thomasm-ttd) [inline src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:900]: How about testing that the update hasn't affected other sites? I know the code currently doesn't have that issue, but is it worth testing that?

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java [hunk 4]
```diff
@@ -388,13 +377,33 @@ private void handleSiteUpdate(RoutingContext rc) {
...
+                existingSite.setName(name);
+            }
 
             uploadSiteToStoreWriterAndWriteExistingSiteToResponse(existingSite, rc);
         } catch (Exception e) {
             rc.fail(500, e);
         }
     }
 
+    private boolean validateSiteName(RoutingContext rc, String name) {
```
Comment (thomasm-ttd) [inline src/main/java/com/uid2/admin/vertx/service/SiteService.java:391]: This feels like the wrong way round to me - I would expect something called validateSiteName to return true if is was valid, and false is it was not.

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java [hunk 5]
```diff
@@ -388,13 +377,33 @@ private void handleSiteUpdate(RoutingContext rc) {
                     ResponseUtil.error(rc, 400, "Invalid parameter for visible: " + visibleParam);
                 }
             }
+            if (name != null) {
+                if (validateSiteName(rc, name)) return;
```
Comment (thomasm-ttd) [inline src/main/java/com/uid2/admin/vertx/service/SiteService.java:381]: It appears the convention in this file (which I also prefer) is that if statements have a following brace: 
if (validateSitenName(rc, name)) {
    return;
}

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java [hunk 6]
```diff
@@ -388,13 +377,33 @@ private void handleSiteUpdate(RoutingContext rc) {
...
+        if (name == null || name.isEmpty()) {
+            ResponseUtil.error(rc, 400, "must specify a valid site name");
+            return true;
+        }
+
+        Optional<Site> existingSite = this.siteProvider.getAllSites()
+                .stream().filter(c -> c.getName().equals(name))
+                .findFirst();
+        if (existingSite.isPresent()) {
+            ResponseUtil.error(rc, 400, "site existed");
```
Comment (alex-yau-ttd) [inline src/main/java/com/uid2/admin/vertx/service/SiteService.java:401]: nit: could we possibly align on some language and do something like we do in [ServiceService.java](https://github.com/IABTechLab/uid2-admin/blob/main/src/main/java/com/uid2/admin/vertx/service/ServiceService.java#L135)

So:
```
ResponseUtil.error(rc, 400, "site with name" + name + " already exists");
```

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java [hunk 7]
```diff
@@ -876,4 +876,63 @@ void addSiteWithDuplicateAppNames(Vertx vertx, VertxTestContext testContext) {
...
+
+        Site[] sites = {
+                new Site(11, "site1", false),
+                new Site(12, "site2", true),
+                new Site(13, "site3", false, Set.of("test1.com", "test2.net")),
+                new Site(14, "site4", false, null, Set.of("test1.com", "test2.net"), Set.of("com.123.game.app.android", "12345678")),
+        };
+        setSites(sites);
+
+        post(vertx, testContext, "api/site/update?id=11&name=site2", null, response -> {
```
Comment (alex-yau-ttd) [inline src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:916]: nit: I think extracting this case to a separate test `renameSiteTestDuplicateName` would help with readability

**Review by thomasm-ttd — APPROVED** [review]

**Review by alex-yau-ttd — APPROVED** [review]

**Pull request merged**